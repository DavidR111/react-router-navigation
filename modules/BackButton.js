/* @flow */
/* eslint global-require: 0 */
/* eslint new-cap: 0 */

import React from 'react'
import { StyleSheet, Platform, View, TouchableNativeFeedback, TouchableOpacity, Image } from 'react-native'

type Props = {
  onPress: Function,
  style?: StyleSheet,
  color?: 'default' | 'light' | 'dark',
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    margin: Platform.OS === 'ios' ? 10 : 16,
  },
  img: {
    height: 24,
    width: 24,
    resizeMode: 'contain',
  },
})

const isAndroid = Platform.OS === 'android'

const Touchable = isAndroid
  ? TouchableNativeFeedback
  : TouchableOpacity

const arrows = {
  default: isAndroid
    ? require('./assets/back-dark.png')
    : require('./assets/back-default.png'),
  light: isAndroid
    ? require('./assets/back-light.png')
    : require('./assets/back-light.png'),
  dark: isAndroid
    ? require('./assets/back-dark.png')
    : require('./assets/back-dark.png'),
}

const BackButton = ({ onPress, style, color = 'default' }: Props): React$Element<any> => {
  const Arrow = arrows[color]
  const background = isAndroid && TouchableNativeFeedback.Ripple('white', true)
  return (
    <Touchable
      style={style}
      background={background}
      onPress={onPress}
    >
      <View style={styles.container}>
        <Image
          style={styles.main}
          source={Arrow}
        />
      </View>
    </Touchable>
  )
}

export default BackButton
