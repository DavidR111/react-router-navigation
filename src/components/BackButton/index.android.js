/* @flow */
/* eslint-disable global-require */
/* eslint-disable new-cap */

import React from 'react'
import { TouchableNativeFeedback, View, Image } from 'react-native'
import type { NavigationTransitionProps } from './../../types'
import styles from './styles'

type Props = NavigationTransitionProps & {
  onPress: () => void,
}

const BackButton = (props: Props) => {
  const { onPress } = props
  const { component } = props.scene.route
  const backButtonStyle = component && (
    component.backButtonStyle || 'dark'
  )
  const backButtonIcons = {
    light: require('./img/back-light.png'),
    dark: require('./img/back-dark.png'),
  }
  const backButtonIcon = backButtonIcons[backButtonStyle]
  return (
    <TouchableNativeFeedback
      onPress={onPress}
      background={TouchableNativeFeedback.Ripple('white', true)}
    >
      <View style={styles.main}>
        <Image source={backButtonIcon} />
      </View>
    </TouchableNativeFeedback>
  )
}

export default BackButton
