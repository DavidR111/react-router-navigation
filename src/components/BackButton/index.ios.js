/* @flow */
/* eslint global-require: 0 */
/* eslint import/newline-after-import: 0 */

import React from 'react'
import { Image, TouchableOpacity } from 'react-native'
import type { NavigationSceneProps } from './../../types'
import styles from './styles'

type Props = NavigationSceneProps & {
  onPress: Function,
}

const BackButton = (props: Props) => {
  const { onPress, scene } = props
  const { component } = scene.route
  const backButtonStyle = component && (
    component.backButtonStyle || 'default'
  )
  const backButtonIcons = {
    default: require('./img/back-default.png'),
    light: require('./img/back-light.png'),
    dark: require('./img/back-dark.png'),
  }
  const backButtonIcon = backButtonIcons[backButtonStyle]
  return (
    <TouchableOpacity style={styles.main} onPress={onPress}>
      <Image source={backButtonIcon} />
    </TouchableOpacity>
  )
}

export default BackButton
