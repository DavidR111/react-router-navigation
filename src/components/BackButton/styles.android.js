/* @flow */

import { StyleSheet } from 'react-native'

const NAVBAR_HEIGHT = 56

const styles = StyleSheet.create({
  main: {
    alignItems: 'center',
    justifyContent: 'center',
    width: NAVBAR_HEIGHT,
    height: NAVBAR_HEIGHT,
  },
})

export default styles
