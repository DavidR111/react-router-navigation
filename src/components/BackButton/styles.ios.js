/* @flow */

import { Platform, StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  button: {
    height: 24,
    width: 24,
    margin: Platform.OS === 'ios' ? 10 : 16,
    resizeMode: 'contain',
  },
})

export default styles
