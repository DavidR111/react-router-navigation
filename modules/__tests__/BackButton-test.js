import React from 'react'
import BackButton from './../BackButton'
import { shallow } from 'enzyme'
import { TouchableOpacity } from 'react-native'
import renderer from 'react-test-renderer'

it('<BackButton /> renders correctly', () => {
  const component = renderer.create(
    <BackButton />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('<BackButton /> calls onPress prop on Press action ', () => {
  const onPress = jest.fn(() => true)
  const tree = shallow(
    <BackButton onPress={onPress} />
  )
  tree.find(TouchableOpacity).first().simulate('press')
  expect(onPress).toBe(true)
})
