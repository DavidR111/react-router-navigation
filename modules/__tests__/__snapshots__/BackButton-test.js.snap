exports[`test <BackButton /> calls onPress prop on Press action  1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID={undefined}>
  <View>
    <Image
      source={1}
      style={
        Object {
          "height": 24,
          "margin": 10,
          "resizeMode": "contain",
          "width": 24,
        }
      } />
  </View>
</View>
`;

exports[`test <BackButton /> renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID={undefined}>
  <View>
    <Image
      source={1}
      style={
        Object {
          "height": 24,
          "margin": 10,
          "resizeMode": "contain",
          "width": 24,
        }
      } />
  </View>
</View>
`;
